tools/modelgen/generators/templates/register.tmpl

```diff
@@ -0,0 +1,5 @@
+{{ range .Objects }}
+func Register{{ initialCap .Name }}Callbacks(handler {{ initialCap .Name }}Callbacks) {
+  objCallbackHandler.{{ initialCap .Name }}Cb = handler
```

Comment:
> so that client is not forced to register callback for objects it doesn't care about.

You don’t have to register for objects you are not interested in. This change went in last week.
You can register for only the objects you are interested in. E.g.

// Register Callbacks
        contivModel.RegisterAppCallbacks(ctrler)
        contivModel.RegisterEndpointGroupCallbacks(ctrler)
        contivModel.RegisterNetworkCallbacks(ctrler)
        contivModel.RegisterPolicyCallbacks(ctrler) -sukhesh


From: Madhav Puri <notifications@github.com<mailto:notifications@github.com>>
Reply-To: contiv/objmodel <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Wednesday, August 19, 2015 at 10:44 AM
To: contiv/objmodel <objmodel@noreply.github.com<mailto:objmodel@noreply.github.com>>
Cc: Sukhesh Halemane <shaleman@cisco.com<mailto:shaleman@cisco.com>>
Subject: Re: [objmodel] Modelgen refactors (cont.) (#2)


In tools/modelgen/generators/templates/register.tmpl<https://github.com/contiv/objmodel/pull/2#discussion_r37444327>:

> @@ -0,0 +1,5 @@
> +{{ range .Objects }}
> +func Register{{ initialCap .Name }}Callbacks(handler {{ initialCap .Name }}Callbacks) {
> +  objCallbackHandler.{{ initialCap .Name }}Cb = handler


yeah, this is returning an error now and may be it can invoke a default handler.

I was suggesting to have a <>Cb initialized to a default no-op handler instead so that client is not forced to register callback for objects it doesn't care about.

This will also help when number of objects and clients grow/change. We ideally don't want to change the clients that don't care whenever that happens.

—
Reply to this email directly or view it on GitHub<https://github.com/contiv/objmodel/pull/2/files#r37444327>.